"""Maps of string IDs to their respective protobuf enums."""

from collections import defaultdict

from katamari_diary.models.katamari_object_pb2 import Category, Location, Size

CATEGORY_ENUM_MAP = defaultdict(
    lambda: Category.INVALID_CATEGORY,
    {
        "OT_CTG_040": Category.DANGER,
        "OT_CTG_063": Category.GUIDANCE,
        "OT_CTG_041": Category.MEASURING,
        "OT_CTG_010": Category.CLEANING,
        "OT_CTG_060": Category.ROADWAYS,
        "OT_CTG_062": Category.ADVERTISING,
        "OT_CTG_058": Category.PARTITIONS,
        "OT_CTG_012": Category.STATIONARY,
        "OT_CTG_016": Category.POWERFUL,
        "OT_CTG_027": Category.GARDENING,
        "OT_CTG_002": Category.VEGETABLES,
        "OT_CTG_050": Category.PLAYGROUND,
        "OT_CTG_025": Category.POST,
        "OT_CTG_072": Category.ADULTS,
        "OT_CTG_075": Category.COUSINS,
        "OT_CTG_042": Category.ART,
        "OT_CTG_059": Category.ENTRANCES_AND_EXITS,
        "OT_CTG_017": Category.PLAYTIME,
        "OT_CTG_015": Category.SPORTS,
        "OT_CTG_052": Category.FARMING,
        "OT_CTG_073": Category.WORKERS,
        "OT_CTG_076": Category.CUTE,
        "OT_CTG_028": Category.FLOWERS,
        "OT_CTG_046": Category.CELEBRATION,
        "OT_CTG_019": Category.CONTAINERS,
        "OT_CTG_044": Category.JAPAN,
        "OT_CTG_064": Category.PLANTS,
        "OT_CTG_018": Category.GAMES,
        "OT_CTG_006": Category.JAPANESE_FOOD,
        "OT_CTG_039": Category.WEAPONS,
        "OT_CTG_011": Category.READING,
        "OT_CTG_009": Category.TRASH,
        "OT_CTG_030": Category.TOOLS,
        "OT_CTG_049": Category.SCHOOL,
        "OT_CTG_008": Category.COOKING,
        "OT_CTG_007": Category.DRINKS,
        "OT_CTG_004": Category.FOOD,
        "OT_CTG_032": Category.POLICE,
        "OT_CTG_013": Category.FASHION,
        "OT_CTG_048": Category.HEROES,
        "OT_CTG_047": Category.EVIL,
        "OT_CTG_045": Category.FESTIVAL,
        "OT_CTG_003": Category.FRUIT,
        "OT_CTG_080": Category.WINGS,
        "OT_CTG_034": Category.RICH,
        "OT_CTG_026": Category.LUGGAGE,
        "OT_CTG_020": Category.DECORATIONS,
        "OT_CTG_055": Category.HOUSES,
        "OT_CTG_029": Category.COMMUNICATION,
        "OT_CTG_066": Category.WHEELS,
        "OT_CTG_057": Category.FACILITIES,
        "OT_CTG_056": Category.STORES,
        "OT_CTG_054": Category.SYMBOLS,
        "OT_CTG_023": Category.ELECTRONICS,
        "OT_CTG_069": Category.ROMANCE,
        "OT_CTG_068": Category.WORK_VEHICLES,
        "OT_CTG_067": Category.TRANSPORT,
        "OT_CTG_031": Category.SUMMER,
        "OT_CTG_021": Category.NECESSITIES,
        "OT_CTG_081": Category.ANIMALS,
        "OT_CTG_005": Category.SNACKS,
        "OT_CTG_033": Category.SCIENCE,
        "OT_CTG_051": Category.ENERGY,
        "OT_CTG_043": Category.CONTROL,
        "OT_CTG_022": Category.RAIN,
        "OT_CTG_014": Category.SOUND,
        "OT_CTG_035": Category.LIGHTING,
        "OT_CTG_079": Category.AQUARIUM,
        "OT_CTG_037": Category.COLD,
        "OT_CTG_024": Category.FURNITURE,
        "OT_CTG_038": Category.SEATING,
        "OT_CTG_070": Category.CHILDREN,
        "OT_CTG_061": Category.VENDING_MACHINES,
        "OT_CTG_036": Category.HOT,
        "OT_CTG_077": Category.LIVESTOCK,
        "OT_CTG_071": Category.TEENAGERS,
        "OT_CTG_078": Category.FIERCE,
        "OT_CTG_074": Category.PROFESSIONALS,
        "OT_CTG_065": Category.NATURE,
        "OT_CTG_053": Category.INDUSTRY,
    },
)

LOCATION_ENUM_MAP = defaultdict(
    lambda: Location.INVALID_LOCATION,
    {
        "OT_MPN_033": Location.LIVING_ROOM,
        "OT_MPN_034": Location.GARDEN_AND_PORCH,
        "OT_MPN_040": Location.N_PIGEON_ST,
        "OT_MPN_038": Location.ENTRY_HALL,
        "OT_MPN_043": Location.S_PIGEON_ST,
        "OT_MPN_052": Location.URCHIN_TOWN,
        "OT_MPN_044": Location.ROADWAY,
        "OT_MPN_049": Location.CONSTRUCTION_SITE,
        "OT_MPN_061": Location.DOLPHIN_CITY,
        "OT_MPN_042": Location.QUAIL_RIVER,
        "OT_MPN_041": Location.SHOPPING_STREET,
        "OT_MPN_050": Location.SEAGULL_PARK,
        "OT_MPN_045": Location.DOVE_LAKE,
        "OT_MPN_037": Location.KIDS_ROOM,
        "OT_MPN_036": Location.KITCHEN,
        "OT_MPN_046": Location.SPARROW_HILL,
        "OT_MPN_057": Location.CIRCUS,
        "OT_MPN_039": Location.HALLWAY,
        "OT_MPN_035": Location.ROOF,
        "OT_MPN_051": Location.PARADISE,
        "OT_MPN_053": Location.SEABREEZE_TOWN,
        "OT_MPN_056": Location.FISHING_PORT,
        "OT_MPN_063": Location.BIG_BLUE_SEA,
        "OT_MPN_047": Location.BASEBALL_FIELD,
        "OT_MPN_055": Location.SUNSHINE_BEACH,
        "OT_MPN_054": Location.SEASLUG_RANCH,
        "OT_MPN_048": Location.CAMPGROUND,
        "OT_MPN_059": Location.WHIRLPOOL_POINT,
        "OT_MPN_058": Location.MULLET_BAY,
        "OT_MPN_060": Location.KUROSHIO_POINT,
        "OT_MPN_062": Location.WHALE_CITY,
    },
)

SIZE_ENUM_MAP = defaultdict(
    lambda: Size.INVALID_SIZE,
    {
        "UI_NIC_009": Size.THE_SMALLEST,
        "UI_NIC_010": Size.EXTREMELY_SMALL,
        "UI_NIC_011": Size.VERY_SMALL,
        "UI_NIC_012": Size.SMALL,
        "UI_NIC_013": Size.SOMEWHAT_SMALL,
        "UI_NIC_014": Size.AVERAGE,
        "UI_NIC_015": Size.ABOVE_AVERAGE,
        "UI_NIC_016": Size.SOMEWHAT_LARGE,
        "UI_NIC_017": Size.LARGE,
        "UI_NIC_018": Size.VERY_LARGE,
        "UI_NIC_019": Size.EXTREMELY_LARGE,
        "UI_NIC_020": Size.THE_LARGEST,
    },
)
